### PR TITLE
[graphite] Make get_table threadsafe

### LINKED ISSUE #922: [graphite] get_table is not threadsafe
Fix this comment:
```
    /* TODO Not thread-safe, but fairly harmless.
     * We can do the double-checked pointer cmpexch thing here. */
```
It's prepending to a prepend-only singly-linked list, so should use pointer cmpexch like we do in other places.

### PR DESCRIPTION
**Pull request by ebraminio**
Fixes #922 hopefully

### CODE REVIEW

File: src/hb-graphite2.cc [hunk 1]
```diff
@@ -79,10 +79,9 @@ static const void *hb_graphite2_get_table (const void *data, unsigned int tag, s
     p->blob = blob;
     p->tag = tag;
 
-    /* TODO Not thread-safe, but fairly harmless.
-     * We can do the double-checked pointer cmpexch thing here. */
-    p->next = face_data->tlist;
-    face_data->tlist = p;
+    p->next = (hb_graphite2_tablelist_t *) hb_atomic_ptr_get (&face_data->tlist);
+    if (!hb_atomic_ptr_cmpexch (&face_data->tlist, nullptr, p))
+      free (p);
```
Comment (behdad) [inline src/hb-graphite2.cc]: We should "goto retry". See other examples of cmpexch use.

**Comment by ebraminio** [comment]
Now I guess I got it right, have a look please

File: src/hb-graphite2.cc [hunk 2]
```diff
@@ -79,10 +80,13 @@ static const void *hb_graphite2_get_table (const void *data, unsigned int tag, s
     p->blob = blob;
     p->tag = tag;
 
-    /* TODO Not thread-safe, but fairly harmless.
-     * We can do the double-checked pointer cmpexch thing here. */
-    p->next = face_data->tlist;
-    face_data->tlist = p;
+    hb_graphite2_tablelist_t *tlist = (hb_graphite2_tablelist_t *) hb_atomic_ptr_get (&face_data->tlist);
```
Comment (ebraminio) [inline src/hb-graphite2.cc]: Or should this be `hb_graphite2_tablelist_t **tlist = (hb_graphite2_tablelist_t **) hb_atomic_ptr_get (&face_data->tlist);`?

**Comment by behdad** [comment]
Any idea why coveralls is being flaky?

**Comment by ebraminio** [comment]
It is saying the newly added code is not covered by the test runs and that's why it is giving a minus score and symbolically failing and I guess we should get to use it :)

**Comment by behdad** [comment]
Can we configure it to not fail if the coverage decrease is small? Or not fail at all?

**Comment by ebraminio** [comment]
Asked on https://github.com/exercism/exercism.io/issues/3139

**Comment by ebraminio** [comment]
Oh its setting is here, https://coveralls.io/github/harfbuzz/harfbuzz/settings and "COVERAGE DECREASE THRESHOLD FOR FAILURE" to 2 percent and disabled its leaving comment option.

**Pull request merged**